packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (!house.owner.equals(userDoc._id)) {
+      throw new HttpException(
+        'user is not the owner of the house',
+        HttpStatus.FORBIDDEN,
+      );
+    }
+
+    this.taskStoreService.delete(task._id);
+  }
+
+  @Put('/:id/completed')
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to mark as complete or not',
+  })
+  @ApiOperation({
+    summary: 'mark user as having completed/not completed task.',
+  })
+  @ApiResponse({
+    description: 'tasks successfully marked as complete/not complete.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not assigned to the task',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async markTaskAsComplete(
+    @Body() completeTaskDto: CompleteTaskDto,
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+  ) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    if (task.assigned !== user.uid) {
+      throw new HttpException(
+        'user is not assigned to task',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    if (completeTaskDto.isComplete) {
+      this.taskStoreService.update(task._id, { lastCompleted: new Date() });
+    } else {
+      this.taskStoreService.update(task._id, { lastCompleted: null });
+    }
+  }
+
+  @Put('/:id')
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id of the task to update',
+  })
+  @ApiOperation({ summary: 'Modify task name, decription or pool' })
+  @ApiResponse({
+    description: 'task successfuly updated.',
+  })
+  @ApiForbiddenResponse({
+    description: 'user is not the owner of the house or not in the house',
+  })
+  @ApiNotFoundResponse({
+    description: 'task does not exist',
+  })
+  async modifyTask(
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+    @Body() updateHouseTasksDto: UpdateHouseTasksDto,
+  ) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException(
+        'user is not in the house',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    const task = await this.taskStoreService.findOne(id);
+    if (task) {
+      const updatedTask = { ...updateHouseTasksDto, assigned: task.assigned };
+
+      if (!house.owner.equals(userDoc._id)) {
+        throw new HttpException(
+          'user is not owner of house',
+          HttpStatus.FORBIDDEN,
+        );
+      }
```

Comment:
same question as above. We could just populate the house object and compare the firebaseId of the owner with our function param